fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/dao/impl/ParameterVisitorBatchDAO.java

```diff
@@ -238,6 +245,12 @@ public void dateValue(String parameterName, Timestamp date, Timestamp dateStart,
             int parameterNameId = getParameterNameId(parameterName);
             
             if (isBase) {
+                // store in the base (resource) table
+                if (logger.isLoggable(Level.FINE)) {
+                    logger.fine("baseDateValue: " + parameterName + "[" + parameterNameId + "], " 
+                            + date + " [" + dateStart + ", " + dateEnd + "]");
+                }
+                
                 // Insert record into the base level date attribute table
                 resourceDates.setInt(1, parameterNameId);
                 resourceDates.setTimestamp(2, date);
```

Comment:
`TIMESTAMP('2007-09-24-15.53.37.2162474',9);`
We could set this as 7 instead of the 9 you found.

Reply:
a couple more examples below like this.

Reply:
sorry, i don't know if i understand the suggestion

Reply:
after further investigation, derby DDL doesn't support specifying TIMESTAMP precision.
still, I've added support for precision for Db2 (and other future impls) to the PR.